### PR TITLE
Mapbox misc

### LINKED ISSUE #74: Map: tooltip
Tooltip/popups -- these each contain an event card vue component -- need to be rendered when a marker is clicked. 

Also on click we should set a "selected event id" in the store so that the events list can highlight / scroll to that item as well (#59 implements the store mutator to do this, so it'll be like, `store.commit('eventSelected', id)`)

### PR DESCRIPTION
**Pull request by stuartsan**
Closes #74, #75, #76

- Updates to the latest mapbox-gl by necessity
- Adds mobile and desktop friendly popups when event markers are clicked
- Zooms into the center of a cluster when the cluster is clicked
- Adds star marker style for official ACLU events

### CODE REVIEW

File: src/components/EventMap.js [hunk 1]
```diff
@@ -115,20 +128,113 @@ export default function(store){
...
+        this.mapRef.on('click', 'unclustered-points', (e) => {
+          store.commit('eventSelected', eventId);
+
+          const eventId = e.features[0].properties.id;
+          const eventCoordinates = e.features[0].geometry.coordinates;
+
+          // Create a Vue instance _inside_ a mapbox Map instance
+          // _inside_ another Vue instance WHOAH. The point is
+          // to reuse the existing event card component.
+          const vm = new Vue({
```
Comment (stuartsan) [inline src/components/EventMap.js]: The length of this method is making me want to cry, maybe I'll split out this part at least :grin:

File: src/components/EventMap.js [hunk 2]
```diff
@@ -115,20 +128,117 @@ export default function(store){
...
+            this.mapRef.getCanvas().style.cursor = 'pointer';
+          });
+
+          this.mapRef.on('mouseleave', layer.id, (e) => {
+            this.mapRef.getCanvas().style.cursor = '';
+          });
+        });
+      },
+
+      openPopupsOnClick() {
```
Comment (stuartsan) [inline src/components/EventMap.js]: [This method needs to be split up a bit or something, it is bananas]

File: src/components/EventMap.js [hunk 3]
```diff
@@ -115,20 +128,117 @@ export default function(store){
...
+              template: '<event-card :event="event"></event-card>',
+              data: {
+                event: eventMap.filteredEvents.find(ev => ev.id === eventId)
+              },
+              components: {
+                'event-card': EventCard
+              }
+            }).$mount();
+
+
```
Comment (tasandberg) [inline src/components/EventMap.js]: Nit - Remove extra empty lines

**Review by tasandberg — CHANGES_REQUESTED** [review]
This is all working really well functionality-wise.  Style-wise the markers are looking odd on my machine - the stars are very tiny and are at the very tip of the map marker, and also the markers seem to be about 3x bigger than in the designs, in relation to other map items.
screen shot 2017-04-08 at 9 23 02 pm

**Comment by stuartsan** [comment]
@Timmehs , hrmmm, would you mind checking if the markers look overly large for you on master/mapstaging too? This will help me troubleshoot it.

File: src/components/EventMap.js [hunk 4]
```diff
@@ -152,8 +152,24 @@ export default function(store){
...
+            event: this.filteredEvents.find(ev => ev.id === eventId)
+          },
+          components: {
+            'event-card': EventCard
+          }
+        }).$mount();
+
+        return vm.$el;
+      },
+
```
Comment (tasandberg) [inline src/components/EventMap.js:171]: 👍

**Comment by stuartsan** [comment]
@Timmehs , I've fixed that bug and confirmed on both retina and non retina screens (guessing you were on retina, this is where I was able to repro). If there's anything else I can adjust for this PR LMK, else I'll give it a merge.

**Pull request merged**